From 892aa70d237aa890eb05244f2d124494bbb65e95 Mon Sep 17 00:00:00 2001
From: Scott Ware <scott.r.ware@intel.com>
Date: Wed, 25 Oct 2017 10:24:06 +0100
Subject: [PATCH] Do not build thin archive

Signed-off-by: Scott Ware <scott.r.ware@intel.com>
---
 common.gypi | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/common.gypi b/common.gypi
index af9c6ba..7a7bb89 100644
--- a/common.gypi
+++ b/common.gypi
@@ -284,8 +291,7 @@
         'cflags_cc': [ '-fno-rtti', '-fno-exceptions', '-std=gnu++0x' ],
         'ldflags': [ '-rdynamic' ],
         'target_conditions': [
-          # The 1990s toolchain on SmartOS can't handle thin archives.
-          ['_type=="static_library" and OS=="solaris"', {
+          ['_type=="static_library"', {
             'standalone_static_library': 1,
           }],
           ['OS=="openbsd"', {
-- 
1.9.1

