From e41c3783d1fdbca9572fcb386c9c16663dc7cc63 Mon Sep 17 00:00:00 2001
From: Scott Ware <scott.r.ware@intel.com>
Date: Thu, 8 Nov 2018 14:08:34 +0000
Subject: [PATCH 2/3] Fix cmake build issues

Upstream-Status: Pending

Signed-off-by: Scott Ware <scott.r.ware@intel.com>
Signed-off-by: Robert Yang <liezhi.yang@windriver.com>
---
 iothub_client/CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/iothub_client/CMakeLists.txt b/iothub_client/CMakeLists.txt
index d04f7bd7..3616b432 100644
--- a/iothub_client/CMakeLists.txt
+++ b/iothub_client/CMakeLists.txt
@@ -332,7 +332,7 @@ if(${use_amqp})
             ${iothub_client_amqp_ws_transport_c_files}
             ${iothub_client_amqp_ws_transport_h_files})
 
-        set(iothub_client_libs ${iothub_client_libs} uamqp)
+        target_link_libraries(${iothub_client_libs} "-luamqp")
     endif(build_as_dynamic)
 endif()
