From 12142ff3508454339af554fa2b0376da0e79f12a Mon Sep 17 00:00:00 2001
From: Scott Ware <scott.r.ware@intel.com>
Date: Fri, 17 Apr 2020 16:09:26 +0100
Subject: [PATCH 2/2] Use installed parson lib

Signed-off-by: Scott Ware <scott.r.ware@intel.com>
---
 CMakeLists.txt     | 17 -----------------
 dependencies.cmake |  3 +++
 2 files changed, 3 insertions(+), 17 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index b74c7a7..7840000 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -130,17 +130,6 @@ endif()
 #Use solution folders.
 set_property(GLOBAL PROPERTY USE_FOLDERS ON)
 
-# build the parson library for json parsing
-add_library(parson
-    ./deps/parson/parson.c
-    ./deps/parson/parson.h
-)
-if(MSVC)
-    set_source_files_properties(../deps/parson/parson.c PROPERTIES COMPILE_FLAGS "/wd4244 /wd4232")
-endif()
-set(parson_h_install_files ./deps/parson/parson.h)
-set(parson_install_libs parson)
-
 if(IN_OPENWRT)
     ADD_DEFINITIONS("$ENV{TARGET_LDFLAGS}" "$ENV{TARGET_CPPFLAGS}" "$ENV{TARGET_CFLAGS}")
     INCLUDE_DIRECTORIES("$ENV{TOOLCHAIN_DIR}/usr/include" "$ENV{TARGET_LDFLAGS}" "$ENV{TARGET_CPPFLAGS}" "$ENV{TARGET_CFLAGS}")
@@ -345,12 +334,6 @@ if(${use_installed_dependencies})
         DESTINATION
             ${package_location}
     )
-else()
-    install(FILES ${parson_h_install_files}
-        DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/azureiot)
-    install(TARGETS ${parson_install_libs}
-        ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}
-        LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR})
 endif()
 
 if (${build_network_e2e})
diff --git a/dependencies.cmake b/dependencies.cmake
index f58bb0b..1698137 100644
--- a/dependencies.cmake
+++ b/dependencies.cmake
@@ -18,6 +18,9 @@ if(${use_installed_dependencies})
         endif ()
     endif ()
 
+    if (NOT parson_FOUND)
+        find_package(parson REQUIRED CONFIG)
+    endif ()
 else ()
     add_subdirectory(c-utility EXCLUDE_FROM_ALL)
 
-- 
2.7.4

