From 82f20163620f5270e0954932f348094f27430543 Mon Sep 17 00:00:00 2001
From: Scott Ware <scott.r.ware@intel.com>
Date: Wed, 8 Mar 2017 13:15:44 +0000
Subject: [PATCH] Skip adding test dependencies if not required

---
 CMakeLists.txt | 14 +++++++++++---
 1 file changed, 11 insertions(+), 3 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 5ca0455..8fef94c 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -88,7 +88,9 @@ if(CMAKE_SYSTEM_NAME STREQUAL "Linux")
   set( MEMORYCHECK_COMMAND_OPTIONS "--trace-children=yes --leak-check=full --suppressions=${CMAKE_CURRENT_LIST_DIR}/core/valgrind_suppressions.txt --error-exitcode=1" )
 endif()
 
-include (CTest)
+if(${run_unittests})
+    include (CTest)
+endif()
 
 include(CheckSymbolExists)
 function(detect_architecture symbol arch)
@@ -178,11 +180,17 @@ macro(compileAsC11)
   endif()
 endmacro(compileAsC11)
 
-enable_testing()
+if(${run_unittests})
+    enable_testing()
+endif()
+
 compileAsC99()
 
 #this adds all test dependencies
-include(dependencies-test.cmake)
+if(${run_unittests})
+    #this adds all test dependencies
+    include(dependencies-test.cmake)
+endif()
 
 #this adds all dependencies
 include(dependencies.cmake)
-- 
1.9.1

