From f07f9d1020c1bfc6d82b4b913d1bbbd3bc9e1156 Mon Sep 17 00:00:00 2001
From: Scott Ware <scott.r.ware@intel.com>
Date: Tue, 17 Jan 2017 20:21:38 +0000
Subject: [PATCH 1/3] Skip adding test dependencies if not required

---
 CMakeLists.txt | 15 +++++++++++----
 1 file changed, 11 insertions(+), 4 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 394da73..2ac06cd 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -66,7 +66,9 @@ if(CMAKE_SYSTEM_NAME STREQUAL "Linux")
   set( MEMORYCHECK_COMMAND_OPTIONS "--trace-children=yes --leak-check=full --suppressions=${CMAKE_CURRENT_LIST_DIR}/core/valgrind_suppressions.txt --error-exitcode=1" )
 endif()
 
-include (CTest)
+if(${run_unittests})
+    include (CTest)
+endif()
 
 include(CheckSymbolExists)
 function(detect_architecture symbol arch) 
@@ -156,11 +158,16 @@ macro(compileAsC11)
   endif()
 endmacro(compileAsC11)
 
-enable_testing()
+if(${run_unittests})
+    enable_testing()
+endif()
+
 compileAsC99()
 
-#this adds all test dependencies
-include(dependencies-test.cmake)
+if(${run_unittests})
+    #this adds all test dependencies
+    include(dependencies-test.cmake)
+endif()
 
 #this adds all dependencies
 include(dependencies.cmake)
-- 
1.9.1

