From 8bcab5e7c887d7875caa276b49986784f70485dc Mon Sep 17 00:00:00 2001
From: Scott Ware <scott.r.ware@intel.com>
Date: Wed, 8 Mar 2017 16:11:22 +0000
Subject: [PATCH 4/4] Remove parson submodule init

---
 dependencies.cmake         | 14 --------------
 modules/dependencies.cmake | 20 --------------------
 2 files changed, 34 deletions(-)

diff --git a/dependencies.cmake b/dependencies.cmake
index d1bb36d..ea9ef2b 100644
--- a/dependencies.cmake
+++ b/dependencies.cmake
@@ -29,17 +29,3 @@ set_platform_files("${CMAKE_CURRENT_LIST_DIR}/deps/c-utility")
 ##########################Find/Install/Build nanomsg###########################
 ###############################################################################
 find_library (NANOMSG_LIB NAMES nanomsg)
-
-###############################################################################
-#############################Init Parson Submodule#############################
-###############################################################################
-if(NOT EXISTS ${PROJECT_SOURCE_DIR}/deps/parson/parson.c)
-    execute_process(
-        COMMAND git submodule update --init ${PROJECT_SOURCE_DIR}/deps/parson
-        WORKING_DIRECTORY ${PROJECT_SOURCE_DIR}
-        RESULT_VARIABLE res   
-    )
-    if(${res})
-        message(FATAL_ERROR "Error pulling parson submodule: ${res}")
-    endif()
-endif()
diff --git a/modules/dependencies.cmake b/modules/dependencies.cmake
index df6fa30..1bfa435 100644
--- a/modules/dependencies.cmake
+++ b/modules/dependencies.cmake
@@ -16,24 +16,4 @@ findAndInstall(umqtt 1.0.25 ${PROJECT_SOURCE_DIR}/deps/umqtt ${PROJECT_SOURCE_DI
 ###############################################################################
 #######################Find/Install/Build azure_iot_sdks#######################
 ###############################################################################
-#The azure_iot_sdks repo requires special treatment. Parson submodule must be initialized.
-if(NOT EXISTS ${PROJECT_SOURCE_DIR}/deps/iot-sdk-c/parson/README.md)
-    execute_process(
-        COMMAND git submodule update --init ${PROJECT_SOURCE_DIR}/deps/iot-sdk-c
-        WORKING_DIRECTORY ${PROJECT_SOURCE_DIR}
-        RESULT_VARIABLE res
-    
-    )
-    if(${res})
-        message(FATAL_ERROR "Error pulling iot-sdk-c submodule: ${res}")
-    endif()
-    execute_process(
-        COMMAND git submodule update --init parson
-        WORKING_DIRECTORY ${PROJECT_SOURCE_DIR}/deps/iot-sdk-c
-        RESULT_VARIABLE res
-    )
-    if(${res})
-        message(FATAL_ERROR "Error pulling parson submodule: ${res}")
-    endif()
-endif()
 findAndInstall(azure_iot_sdks 1.1.5 ${PROJECT_SOURCE_DIR}/deps/iot-sdk-c ${PROJECT_SOURCE_DIR}/deps/iot-sdk-c -Duse_installed_dependencies=ON -Duse_openssl=OFF -Dbuild_as_dynamic=ON -Drun_e2e_tests=ON -Dskip_samples=ON -G "${CMAKE_GENERATOR}")
-- 
1.9.1

