From 8d90b8fa13a2b35635f4374e24506b9c8ef3adc4 Mon Sep 17 00:00:00 2001
From: Scott Ware <scott.r.ware@intel.com>
Date: Tue, 24 Jan 2017 15:34:04 +0000
Subject: [PATCH 6/6] Remove parson submodule init

---
 dependencies.cmake | 14 --------------
 1 file changed, 14 deletions(-)

diff --git a/dependencies.cmake b/dependencies.cmake
index 3642582..5aeeafe 100644
--- a/dependencies.cmake
+++ b/dependencies.cmake
@@ -22,17 +22,3 @@ set_platform_files("${CMAKE_CURRENT_LIST_DIR}/deps/c-utility")
 ##########################Find/Install/Build nanomsg###########################
 ###############################################################################
 find_library (NANOMSG_LIB NAMES nanomsg)
-
-###############################################################################
-#############################Init Parson Submodule#############################
-###############################################################################
-if(NOT EXISTS ${PROJECT_SOURCE_DIR}/deps/parson/parson.c)
-    execute_process(
-        COMMAND git submodule update --init ${PROJECT_SOURCE_DIR}/deps/parson
-        WORKING_DIRECTORY ${PROJECT_SOURCE_DIR}
-        RESULT_VARIABLE res   
-    )
-    if(${res})
-        message(FATAL_ERROR "Error pulling parson submodule: ${res}")
-    endif()
-endif()
-- 
1.9.1

From 44028bd6be7781f1a4679aa7bd3dbbf98a312447 Mon Sep 17 00:00:00 2001
From: Scott Ware <scott.r.ware@intel.com>
Date: Tue, 24 Jan 2017 15:48:44 +0000
Subject: [PATCH 7/7] Remove parson submodule init

---
 modules/dependencies.cmake | 20 --------------------
 1 file changed, 20 deletions(-)

diff --git a/modules/dependencies.cmake b/modules/dependencies.cmake
index b7aae45..eb45577 100644
--- a/modules/dependencies.cmake
+++ b/modules/dependencies.cmake
@@ -16,24 +16,4 @@ findAndInstall(umqtt ${PROJECT_SOURCE_DIR}/deps/umqtt ${PROJECT_SOURCE_DIR}/deps
 ###############################################################################
 #######################Find/Install/Build azure_iot_sdks#######################
 ###############################################################################
-#The azure_iot_sdks repo requires special treatment. Parson submodule must be initialized.
-if(NOT EXISTS ${PROJECT_SOURCE_DIR}/deps/iot-sdk-c/parson/README.md)
-    execute_process(
-        COMMAND git submodule update --init ${PROJECT_SOURCE_DIR}/deps/iot-sdk-c
-        WORKING_DIRECTORY ${PROJECT_SOURCE_DIR}
-        RESULT_VARIABLE res
-    
-    )
-    if(${res})
-        message(FATAL_ERROR "Error pulling iot-sdk-c submodule: ${res}")
-    endif()
-    execute_process(
-        COMMAND git submodule update --init parson
-        WORKING_DIRECTORY ${PROJECT_SOURCE_DIR}/deps/iot-sdk-c
-        RESULT_VARIABLE res
-    )
-    if(${res})
-        message(FATAL_ERROR "Error pulling parson submodule: ${res}")
-    endif()
-endif()
 findAndInstall(azure_iot_sdks ${PROJECT_SOURCE_DIR}/deps/iot-sdk-c ${PROJECT_SOURCE_DIR}/deps/iot-sdk-c -Duse_installed_dependencies=ON -Duse_openssl=OFF -Dbuild_as_dynamic=ON -Drun_e2e_tests=ON -G "${CMAKE_GENERATOR}")
-- 
1.9.1

