From e223760f87cca46202cfed33c7a214b6e331215a Mon Sep 17 00:00:00 2001
From: Scott Ware <scott.r.ware@intel.com>
Date: Fri, 24 Mar 2017 12:40:55 +0000
Subject: [PATCH] Generalise authorisation message

---
 src/auth_web.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/auth_web.py b/src/auth_web.py
index f57a9ef..069c864 100755
--- a/src/auth_web.py
+++ b/src/auth_web.py
@@ -65,7 +65,7 @@ class Start(object):
 
 		return (
 			"<h2>Success!</h2><h3> Refresh token has been added to your "
-			"config file, you may now reboot the Pi </h3><br>{}"
+			"config file. </h3><br>{}"
 		).format(resp['refresh_token'])
 
 	index.exposed = True
-- 
1.9.1

