From 06c4a11627454b97d3edc25d985d599541164742 Mon Sep 17 00:00:00 2001
From: Scott Ware <scott.r.ware@intel.com>
Date: Mon, 3 Apr 2017 17:32:49 +0100
Subject: [PATCH 2/2] Use HTTPS for authentication

---
 src/auth_web.py | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/src/auth_web.py b/src/auth_web.py
index 069c864..ae7218b 100755
--- a/src/auth_web.py
+++ b/src/auth_web.py
@@ -71,12 +71,15 @@ class Start(object):
 	index.exposed = True
 	code.exposed = True
 
+cherrypy.server.ssl_module = 'builtin'
+cherrypy.server.ssl_certificate = "/etc/opt/AlexaPi/cert.pem"
+cherrypy.server.ssl_private_key = "/etc/opt/AlexaPi/privkey.pem"
 cherrypy.config.update({'server.socket_host': '0.0.0.0'})
 cherrypy.config.update({'server.socket_port': int(os.environ.get('PORT', '5050'))})
 cherrypy.config.update({"environment": "embedded"})
 
 
 ip = [(s.connect(('8.8.8.8', 53)), s.getsockname()[0], s.close()) for s in [socket.socket(socket.AF_INET, socket.SOCK_DGRAM)]][0][1]
-print("Ready goto http://{}:5050 or http://localhost:5050  to begin the auth process".format(ip))
+print("Ready goto https://{}:5050 to begin the auth process".format(ip))
 print("(Press Ctrl-C to exit this script once authorization is complete)")
 cherrypy.quickstart(Start())
-- 
1.9.1

