From 2463c218c794655c1068fbd467949c5ee19de546 Mon Sep 17 00:00:00 2001
From: Scott Ware <scott.r.ware@intel.com>
Date: Wed, 30 May 2018 16:56:52 +0100
Subject: [PATCH] Fix packaging issues

Signed-off-by: Scott Ware <scott.r.ware@intel.com>
---
 CMakeLists.txt | 11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 05cba68..dd73357 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -6,6 +6,9 @@ project(uamqp)
 
 FILE(READ ${CMAKE_CURRENT_LIST_DIR}/version.txt UAMQP_VERSION)
 
+set(GENERIC_LIB_VERSION ${UAMQP_VERSION})
+string(SUBSTRING ${UAMQP_VERSION} 0 1 GENERIC_LIB_SOVERSION)
+
 # Include the common build rules for the C SDK
 include(deps/azure-c-shared-utility/configs/azure_iot_build_rules.cmake)
 
@@ -212,6 +215,14 @@ setTargetBuildProperties(uamqp)
 
 target_link_libraries(uamqp aziotsharedutil)
 
+if (NOT WIN32)
+    set_target_properties(uamqp
+        PROPERTIES
+        VERSION ${GENERIC_LIB_VERSION}
+        SOVERSION ${GENERIC_LIB_SOVERSION}
+    )
+endif()
+
 if (NOT ${skip_samples})
     add_subdirectory(samples)
 endif()
-- 
2.7.4

