From 5e901f3644a258eeb4a03521ead0d8153526c841 Mon Sep 17 00:00:00 2001
From: Scott Ware <scott.r.ware@intel.com>
Date: Thu, 8 Nov 2018 12:47:26 +0000
Subject: [PATCH] Hardcode c shared utility path

Signed-off-by: Scott Ware <scott.r.ware@intel.com>
---
 CMakeLists.txt | 1 +----
 1 file changed, 1 insertions(+), 4 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 96a2692..1bcc1f6 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -48,10 +51,7 @@ if(NOT ${use_installed_dependencies})
 endif()
 include("dependencies.cmake")
 
-# Include the common build rules for C shared utility
-include(${SHARED_UTIL_FOLDER}/configs/azure_iot_build_rules.cmake)
-
-set_platform_files(${SHARED_UTIL_FOLDER})
+include(deps/azure-c-shared-utility/configs/azure_iot_build_rules.cmake)
 
 set(run_e2e_tests ${original_run_e2e_tests})
 set(run_int_tests ${original_run_int_tests})
-- 
2.7.4

