From 836c2b6dcf2438ef35e3018ab15f299e76a4e3f9 Mon Sep 17 00:00:00 2001
From: Scott Ware <scott.r.ware@intel.com>
Date: Mon, 21 Nov 2016 17:43:33 +0000
Subject: [PATCH] Skip building dependencies if provided

---
 CMakeLists.txt     |  4 ++--
 dependencies.cmake | 15 +++++++++++----
 2 files changed, 13 insertions(+), 6 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index ae6460d..51420fa 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -220,7 +220,7 @@ add_library(uamqp
     ${socketlistener_c_files}
     )
 
-target_link_libraries(uamqp aziotsharedutil)
+target_link_libraries(uamqp ${SHARED_UTIL_LIB_PATH})
 
 if (NOT ${ARCHITECTURE} STREQUAL "ARM")
     if (NOT ${skip_samples})
@@ -279,4 +279,4 @@ if(${use_installed_dependencies})
 
 else()
     message(WARNING "This package may only be installed when 'use_installed_dependencies' is ON")
-endif()
\ No newline at end of file
+endif()
diff --git a/dependencies.cmake b/dependencies.cmake
index e21dffc..5f09a9e 100644
--- a/dependencies.cmake
+++ b/dependencies.cmake
@@ -1,8 +1,15 @@
 #Copyright (c) Microsoft. All rights reserved.
 #Licensed under the MIT license. See LICENSE file in the project root for full license information.
 
-if(${use_installed_dependencies})
-    find_package(azure_c_shared_utility REQUIRED config)
-else()
+find_library (SHARED_UTIL_LIB_PATH NAMES aziotsharedutil)
+
+# Check if shared utils library is already provided
+if (SHARED_UTIL_LIB_PATH AND AZURE_INCLUDE_DIR)
+    set(SHARED_UTIL_FOUND 1)
+    set(SHARED_UTIL_INC_FOLDER ${AZURE_INCLUDE_DIR})
+endif()
+
+if (NOT SHARED_UTIL_FOUND)
     add_subdirectory(c-utility)
-endif()
\ No newline at end of file
+    set(SHARED_UTIL_LIB_PATH aziotsharedutil)
+endif()
-- 
1.9.1

