From ecf89e11d0f374fe8ef9d9002109f8e29a742ddb Mon Sep 17 00:00:00 2001
From: Scott Ware <scott.r.ware@intel.com>
Date: Thu, 22 Feb 2018 17:50:21 +0000
Subject: [PATCH 1/3] Fix packaging issues

Signed-off-by: Scott Ware <scott.r.ware@intel.com>
---
 CMakeLists.txt | 23 +++++++++++++++++++++++
 1 file changed, 23 insertions(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index d820d15..d48c68d 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -7,6 +7,9 @@ project(uhttp)
 
 set(UHTTP_VERSION 1.0.1)
 
+set(GENERIC_LIB_VERSION ${UHTTP_VERSION})
+string(SUBSTRING ${UHTTP_VERSION} 0 1 GENERIC_LIB_SOVERSION)
+
 option(run_e2e_tests "set run_e2e_tests to ON to run e2e tests (default is OFF)" OFF)
 option(run_unittests "set run_unittests to ON to run unittests (default is OFF)" OFF)
 option(skip_samples "set skip_samples to ON to skip building samples (default is OFF)[if possible, they are always built]" OFF)
@@ -87,6 +90,13 @@ include("configs/azure_uhttpFunctions.cmake")
 
 add_library(uhttp ${uhttp_c_files} ${uhttp_h_files})
 target_link_libraries(uhttp aziotsharedutil)
+if (NOT WIN32)
+    set_target_properties(uhttp
+        PROPERTIES
+        VERSION ${GENERIC_LIB_VERSION}
+        SOVERSION ${GENERIC_LIB_SOVERSION}
+    )
+endif()
 
 set_platform_files(${CMAKE_CURRENT_LIST_DIR}/deps/c-utility)
 
@@ -122,9 +132,22 @@ if(${use_installed_dependencies})
         VERSION ${UHTTP_VERSION}
         COMPATIBILITY SameMajorVersion
     )
+    
+    configure_file("configs/${PROJECT_NAME}Config.cmake"
+        "${CMAKE_CURRENT_BINARY_DIR}/${PROJECT_NAME}/${PROJECT_NAME}Config.cmake"
+        COPYONLY
+    )
+    
+    install(EXPORT uhttpTargets
+        FILE
+            "${PROJECT_NAME}Targets.cmake"
+        DESTINATION
+            ${package_location}
+    )
 
     install(
         FILES
+            "configs/${PROJECT_NAME}Config.cmake"
             "${CMAKE_CURRENT_BINARY_DIR}/${PROJECT_NAME}/${PROJECT_NAME}ConfigVersion.cmake"
         DESTINATION
             ${package_location}
-- 
1.9.1

From 44a34e2c98f2272f540dde7927966b5d62924657 Mon Sep 17 00:00:00 2001
From: Scott Ware <scott.r.ware@intel.com>
Date: Thu, 22 Feb 2018 18:11:31 +0000
Subject: [PATCH 2/2] Add config file

Signed-off-by: Scott Ware <scott.r.ware@intel.com>
---
 configs/uhttpConfig.cmake | 8 ++++++++
 1 file changed, 8 insertions(+)
 create mode 100644 configs/uhttpConfig.cmake

diff --git a/configs/uhttpConfig.cmake b/configs/uhttpConfig.cmake
new file mode 100644
index 0000000..c4e9e91
--- /dev/null
+++ b/configs/uhttpConfig.cmake
@@ -0,0 +1,8 @@
+#Copyright (c) Microsoft. All rights reserved.
+#Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+include("${CMAKE_CURRENT_LIST_DIR}/uhttpTargets.cmake")
+
+get_target_property(UHTTP_INCLUDES uhttp INTERFACE_INCLUDE_DIRECTORIES)
+
+set(UHTTP_INCLUDES ${UHTTP_INCLUDES} CACHE INTERNAL "")
-- 
1.9.1
