From c60575c8a9309fc11aa22fee53318a4da18d62fd Mon Sep 17 00:00:00 2001
From: Scott Ware <scott.r.ware@intel.com>
Date: Thu, 22 Feb 2018 18:04:17 +0000
Subject: [PATCH 2/2] Remove calls to redundant functions

Signed-off-by: Scott Ware <scott.r.ware@intel.com>
---
 CMakeLists.txt | 2 --
 1 file changed, 2 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 6edbe04..4235990 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -43,8 +43,6 @@ if(MSVC)
     set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} /W4")
 endif()
 
-compileAsC99()
-
 set(UHTTP_C_INC_FOLDER ${CMAKE_CURRENT_LIST_DIR}/inc CACHE INTERNAL "this is what needs to be included if using sharedLib lib" FORCE)
 
 #do not add or build any tests of the dependencies
-- 
1.9.1

