From f1f44ca84e17691f1538de2353ef4657c4187c23 Mon Sep 17 00:00:00 2001
From: Scott Ware <scott.r.ware@intel.com>
Date: Wed, 4 Jan 2017 12:10:03 +0000
Subject: [PATCH 3/4] Add option for running tests

---
 CMakeLists.txt | 8 +++++---
 1 file changed, 5 insertions(+), 3 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 62166db..dbd1d2c 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -10,6 +10,7 @@ if(${CMAKE_SOURCE_DIR} STREQUAL ${CMAKE_BINARY_DIR})
 endif()
 
 option(BUILD_DOCS "Create HTML based API documentation (requires Doxygen)" OFF)
+option(RUN_TESTS "Run unit and integration tests" OFF)
 
 ########################################
 # Section : Common SDK Build setttings #
@@ -121,9 +122,10 @@ if(UNIX)
 	add_subdirectory(cli)
 endif()
 
-add_subdirectory(tests/integration)
-
-add_subdirectory(tests/unit)
+if(RUN_TESTS)
+	add_subdirectory(tests/integration)
+	add_subdirectory(tests/unit)
+endif()
 
 add_subdirectory(samples/PubSub)
 
-- 
1.9.1

