From cd309f198c3b97f1d2c275bcbbb349073132a5bf Mon Sep 17 00:00:00 2001
From: Scott Ware <scott.r.ware@intel.com>
Date: Thu, 8 Jun 2017 19:03:42 +0100
Subject: [PATCH 3/3] Add option to disable tests

---
 CMakeLists.txt | 8 +++++---
 1 file changed, 5 insertions(+), 3 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 858b2c8..5738e5d 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -11,6 +11,7 @@ if(${PROJECT_SOURCE_DIR} STREQUAL ${PROJECT_BINARY_DIR})
 endif()
 
 option(BUILD_DOCS "Create HTML based API documentation (requires Doxygen)" OFF)
+option(RUN_TESTS "Run unit and integration tests" OFF)
 
 ########################################
 # Section : Common SDK Build setttings #
@@ -131,9 +132,10 @@ if(UNIX)
 	add_subdirectory(cli)
 endif()
 
-add_subdirectory(tests/integration)
-
-add_subdirectory(tests/unit)
+if(RUN_TESTS)
+	add_subdirectory(tests/integration)
+	add_subdirectory(tests/unit)
+endif()
 
 add_subdirectory(samples/PubSub)
 
-- 
1.9.1

