From f213cbc3a8ae8554d706e408592ba3b664e21fcd Mon Sep 17 00:00:00 2001
From: Scott Ware <scott.r.ware@intel.com>
Date: Fri, 9 Jun 2017 09:29:40 +0100
Subject: [PATCH 4/4] Fix cli includes

Signed-off-by: Scott Ware <scott.r.ware@intel.com>
---
 cli/cli.hpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/cli/cli.hpp b/cli/cli.hpp
index 4f3634c..f0f1ee2 100644
--- a/cli/cli.hpp
+++ b/cli/cli.hpp
@@ -26,7 +26,7 @@
 #include <direct.h>
 #define getcwd _getcwd // avoid MSFT "deprecation" warning
 #else
-#include <limits>
+#include <linux/limits.h>
 #define MAX_PATH_LENGTH_ PATH_MAX
 #endif
 
-- 
1.9.1

