From fe0aa081845caad2b0b7f6de4df564c4412f728e Mon Sep 17 00:00:00 2001
From: Scott Ware <scott.r.ware@intel.com>
Date: Mon, 21 Aug 2017 11:00:35 +0100
Subject: [PATCH 4/7] Add option to build samples

Signed-off-by: Scott Ware <scott.r.ware@intel.com>
---
 CMakeLists.txt | 17 ++++++++---------
 1 file changed, 8 insertions(+), 9 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 5738e5d..e814963 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -11,6 +11,7 @@ if(${PROJECT_SOURCE_DIR} STREQUAL ${PROJECT_BINARY_DIR})
 endif()
 
 option(BUILD_DOCS "Create HTML based API documentation (requires Doxygen)" OFF)
+option(BUILD_SAMPLES "Build samples" OFF)
 option(RUN_TESTS "Run unit and integration tests" OFF)
 
 ########################################
@@ -137,15 +138,13 @@ if(RUN_TESTS)
 	add_subdirectory(tests/unit)
 endif()
 
-add_subdirectory(samples/PubSub)
-
-add_subdirectory(samples/ShadowDelta)
-
-add_subdirectory(samples/Discovery EXCLUDE_FROM_ALL)
-
-add_subdirectory(samples/StoryRobotArm EXCLUDE_FROM_ALL)
-
-add_subdirectory(samples/StorySwitch EXCLUDE_FROM_ALL)
+if(BUILD_SAMPLES)
+	add_subdirectory(samples/PubSub)
+	add_subdirectory(samples/ShadowDelta)
+	add_subdirectory(samples/Discovery EXCLUDE_FROM_ALL)
+	add_subdirectory(samples/StoryRobotArm EXCLUDE_FROM_ALL)
+	add_subdirectory(samples/StorySwitch EXCLUDE_FROM_ALL)
+endif()
 
 ##################################
 # Section: Define Install Target #
-- 
1.9.1

