From a0bd0d6ed249da5cd86af9154c53993734a816d5 Mon Sep 17 00:00:00 2001
From: Scott Ware <scott.r.ware@intel.com>
Date: Thu, 15 Oct 2020 15:11:05 +0100
Subject: [PATCH] Fix packaging issues

Signed-off-by: Scott Ware <scott.r.ware@intel.com>
---
 CMakeLists.txt | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 5175793..89df9d9 100755
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1,5 +1,5 @@
 cmake_minimum_required (VERSION 3.0)
-project (s2n C)
+project (s2n VERSION 1.0.0 LANGUAGES C)
 
 set(CMAKE_ARCHIVE_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/lib)
 set(CMAKE_LIBRARY_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/lib)
@@ -189,7 +189,7 @@ file(GLOB S2N_SRC
 )
 
 add_library(${PROJECT_NAME} ${S2N_HEADERS} ${S2N_SRC})
-set_target_properties(${PROJECT_NAME} PROPERTIES LINKER_LANGUAGE C)
+set_target_properties(${PROJECT_NAME} PROPERTIES LINKER_LANGUAGE C SOVERSION ${PROJECT_VERSION_MAJOR} VERSION ${PROJECT_VERSION})
 
 set(CMAKE_C_FLAGS_DEBUGOPT "")
 
-- 
2.7.4

