From d7f9e9b85dec182c6b1893c3577bab23a3a7ac19 Mon Sep 17 00:00:00 2001
From: Scott Ware <scott.r.ware@intel.com>
Date: Wed, 18 Jan 2017 12:46:21 +0000
Subject: [PATCH 2/3] Remove unit tests

---
 device/iothub_client_python/CMakeLists.txt | 1 -
 1 file changed, 1 deletion(-)

diff --git a/device/iothub_client_python/CMakeLists.txt b/device/iothub_client_python/CMakeLists.txt
index 96e51a5..f207db1 100644
--- a/device/iothub_client_python/CMakeLists.txt
+++ b/device/iothub_client_python/CMakeLists.txt
@@ -85,4 +85,3 @@ if(${use_wsio})
 endif()
 
 add_subdirectory(src)
-add_subdirectory(test)
-- 
1.9.1

