From 67e2b5b5b37f34f0f18ef46acd8c8199d8b2e94b Mon Sep 17 00:00:00 2001
From: Scott Ware <scott.r.ware@intel.com>
Date: Thu, 5 Jul 2018 17:37:06 +0100
Subject: [PATCH] Fix for GCC 8.x

Signed-off-by: Scott Ware <scott.r.ware@intel.com>
---
 src/crt_abstractions.c | 4 ++--
 2 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/c/c-utility/src/crt_abstractions.c b/c/c-utility/src/crt_abstractions.c
index 1a937c0..9dd8688 100644
--- a/c/c-utility/src/crt_abstractions.c
+++ b/c/c-utility/src/crt_abstractions.c
@@ -146,7 +146,7 @@ int strncpy_s(char* dst, size_t dstSizeInBytes, const char* src, size_t maxCount
             }
             else
             {
-                (void)strncpy(dst, src, srcLength);
+                (void)memcpy(dst, src, srcLength);
                 dst[srcLength] = '\0';
                 /*Codes_SRS_CRT_ABSTRACTIONS_99_018: [strncpy_s shall return Zero upon success]*/
                 result = 0;
@@ -160,7 +160,7 @@ int strncpy_s(char* dst, size_t dstSizeInBytes, const char* src, size_t maxCount
                 srcLength = dstSizeInBytes - 1;
                 truncationFlag = 1;
             }
-            (void)strncpy(dst, src, srcLength);
+            (void)memcpy(dst, src, srcLength);
             dst[srcLength] = '\0';
             result = 0;
         }
--

diff --git a/c/serializer/src/commanddecoder.c b/c/serializer/src/commanddecoder.c
index 1a937c0..9dd8688 100644
--- a/c/serializer/src/commanddecoder.c
+++ b/c/serializer/src/commanddecoder.c
@@ -191,7 +191,7 @@ 
 #ifdef _MSC_VER
 #pragma warning(suppress: 6324) /* We intentionally use here strncpy */ 
 #endif
-        if (strncpy(tempStr, actionName, strLength - 1) == NULL)
+        if (memcpy(tempStr, actionName, strLength - 1) == NULL)
         {
             /* Codes_SRS_COMMAND_DECODER_99_021:[ If the parsing of the command fails for any other reason the command shall not be dispatched.] */
             LogError("Invalid action name.");
-- 
2.7.4

