From 6b8e9fa80e3d48f26f1d3d0a804ef5b43d505e4d Mon Sep 17 00:00:00 2001
From: Scott Ware <scott.r.ware@intel.com>
Date: Fri, 3 Feb 2017 10:36:37 +0000
Subject: [PATCH] Fix install requires in setup script

---
 setup.py | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/setup.py b/setup.py
index c14f774..a4a5919 100644
--- a/setup.py
+++ b/setup.py
@@ -49,8 +49,8 @@ setup(
         'License :: OSI Approved :: MIT License',
         'Topic :: Software Development'],
     install_requires=[
-        "msrest~=0.4.4",
+        "msrest==0.4.4",
         "keyring>=5.6",
-        "adal~=0.4.0"
+        "adal>=0.4.0"
     ],
 )
-- 
1.9.1

