From 93fbedadeba467482d79d3b526a5dd838ce2219a Mon Sep 17 00:00:00 2001
From: Scott Ware <scott.r.ware@intel.com>
Date: Tue, 20 Oct 2020 11:06:46 +0100
Subject: [PATCH] Do not override libcrypto location

Signed-off-by: Scott Ware <scott.r.ware@intel.com>
---
 setup.py                 | 2 --
 1 files changed, 2 deletions(-)

diff --git a/setup.py b/setup.py
index a4196dd..c0098b1 100644
--- a/setup.py
+++ b/setup.py
@@ -111,8 +111,6 @@ DEP_BUILD_DIR = os.path.join(PROJECT_DIR, 'build', 'deps')
 DEP_INSTALL_PATH = os.environ.get('AWS_C_INSTALL', os.path.join(DEP_BUILD_DIR, 'install'))
 
 AWS_LIBCRYPTO_INSTALL = None
-if sys.platform != 'darwin' and sys.platform != 'win32':
-    AWS_LIBCRYPTO_INSTALL = os.environ.get('AWS_LIBCRYPTO_INSTALL', os.path.join(DEP_BUILD_DIR, 'libcrypto'))
 
 
 class awscrt_build_ext(setuptools.command.build_ext.build_ext):
