From 7d45ca6436479335c00738c0c18fbf5a71e65305 Mon Sep 17 00:00:00 2001
From: Scott Ware <scott.r.ware@intel.com>
Date: Mon, 21 Nov 2016 17:32:53 +0000
Subject: [PATCH] Skip building dependencies if provided

---
 CMakeLists.txt     |  2 +-
 dependencies.cmake | 16 ++++++++++++----
 2 files changed, 13 insertions(+), 5 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index d412bf2..450bf37 100755
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -159,7 +159,7 @@ add_library(umqtt ${source_c_files} ${source_h_files})
 
 set(SHARED_UTIL_ADAPTER_FOLDER "${CMAKE_CURRENT_LIST_DIR}/c-utility/adapters")
 
-target_link_libraries(umqtt aziotsharedutil)
+target_link_libraries(umqtt ${SHARED_UTIL_LIB_PATH})
 
 if (NOT ${ARCHITECTURE} STREQUAL "ARM")
     if (NOT ${skip_samples})
diff --git a/dependencies.cmake b/dependencies.cmake
index 1148e38..ab4af3f 100644
--- a/dependencies.cmake
+++ b/dependencies.cmake
@@ -1,8 +1,16 @@
 #Copyright (c) Microsoft. All rights reserved.
 #Licensed under the MIT license. See LICENSE file in the project root for full license information.
 
-if(${use_installed_dependencies})
-    find_package(azure_c_shared_utility REQUIRED CONFIG)
-else()
+# Check if shared utils library is already provided
+find_library (SHARED_UTIL_LIB_PATH NAMES aziotsharedutil)
+
+if (SHARED_UTIL_LIB_PATH AND AZURE_INCLUDE_DIR)
+    set(SHARED_UTIL_FOUND 1)
+    set(SHARED_UTIL_INC_FOLDER ${AZURE_INCLUDE_DIR})
+endif()
+
+# Build shared util library if not found
+if (NOT SHARED_UTIL_FOUND)
     add_subdirectory(c-utility)
-endif()
\ No newline at end of file
+    set(SHARED_UTIL_LIB_PATH aziotsharedutil)
+endif()
-- 
1.9.1

