From b2199da2fce59d6d6faf698e203fa3c5890594f6 Mon Sep 17 00:00:00 2001
From: Scott Ware <scott.r.ware@intel.com>
Date: Tue, 15 Nov 2016 17:47:25 +0000
Subject: [PATCH 2/2] Use GNU install directories

---
 CMakeLists.txt | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 60b7397..acac610 100755
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -179,8 +179,10 @@ if (NOT ${skip_unittests})
     add_subdirectory(tests)
 endif()
 
+include(GNUInstallDirs)
+
 if(WIN32)  
 else()  
-    install (TARGETS umqtt DESTINATION lib)  
-    install (FILES ${source_h_files} DESTINATION include/azureiot/azure_umqtt_c)  
+    install (TARGETS umqtt DESTINATION ${CMAKE_INSTALL_LIBDIR})  
+    install (FILES ${source_h_files} DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/azureiot/azure_umqtt_c)  
 endif (WIN32)  
-- 
1.9.1

