From 03432422a2e79ab8267115e06942504efba0177b Mon Sep 17 00:00:00 2001
From: Scott Ware <scott.r.ware@intel.com>
Date: Wed, 23 Aug 2017 16:13:44 +0100
Subject: [PATCH 1/6] Skip adding test dependencies if not required

Signed-off-by: Scott Ware <scott.r.ware@intel.com>
---
 CMakeLists.txt | 15 +++++++++++----
 1 file changed, 11 insertions(+), 4 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 5429488..e856fdf 100755
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -100,7 +100,9 @@ if(CMAKE_SYSTEM_NAME STREQUAL "Linux")
   set( MEMORYCHECK_COMMAND_OPTIONS "--trace-children=yes --leak-check=full --suppressions=${CMAKE_CURRENT_LIST_DIR}/core/valgrind_suppressions.txt --error-exitcode=1" )
 endif()
 
-include (CTest)
+if(${run_unittests})
+    include (CTest)
+endif()
 
 include(CheckSymbolExists)
 function(detect_architecture symbol arch)
@@ -200,11 +202,16 @@ macro(compileAsC11)
   endif()
 endmacro(compileAsC11)
 
-enable_testing()
+if(${run_unittests})
+    enable_testing()
+endif()
+
 compileAsC99()
 
-#this adds all test dependencies
-include(dependencies-test.cmake)
+if(${run_unittests})
+    #this adds all test dependencies
+    include(dependencies-test.cmake)
+endif()
 
 #this adds all dependencies
 include(dependencies.cmake)
-- 
1.9.1

