From de8747c61e75bc64532ecaedc9775d306784b4e1 Mon Sep 17 00:00:00 2001
From: Scott Ware <scott.r.ware@intel.com>
Date: Wed, 23 Aug 2017 16:15:26 +0100
Subject: [PATCH 3/6] Skip parson submodule init

Signed-off-by: Scott Ware <scott.r.ware@intel.com>
---
 modules/dependencies.cmake | 20 --------------------
 1 file changed, 20 deletions(-)

diff --git a/modules/dependencies.cmake b/modules/dependencies.cmake
index 33b8a3d..1f7890f 100644
--- a/modules/dependencies.cmake
+++ b/modules/dependencies.cmake
@@ -16,24 +16,4 @@ findAndInstall(umqtt 1.0.25 ${PROJECT_SOURCE_DIR}/deps/umqtt ${PROJECT_SOURCE_DI
 ###############################################################################
 #######################Find/Install/Build azure_iot_sdks#######################
 ###############################################################################
-#The azure_iot_sdks repo requires special treatment. Parson submodule must be initialized.
-if(NOT EXISTS ${PROJECT_SOURCE_DIR}/deps/iot-sdk-c/parson/README.md)
-    execute_process(
-        COMMAND git submodule update --init ${PROJECT_SOURCE_DIR}/deps/iot-sdk-c
-        WORKING_DIRECTORY ${PROJECT_SOURCE_DIR}
-        RESULT_VARIABLE res
-    
-    )
-    if(${res})
-        message(FATAL_ERROR "Error pulling iot-sdk-c submodule: ${res}")
-    endif()
-    execute_process(
-        COMMAND git submodule update --init parson
-        WORKING_DIRECTORY ${PROJECT_SOURCE_DIR}/deps/iot-sdk-c
-        RESULT_VARIABLE res
-    )
-    if(${res})
-        message(FATAL_ERROR "Error pulling parson submodule: ${res}")
-    endif()
-endif()
 findAndInstall(azure_iot_sdks 1.1.5 ${PROJECT_SOURCE_DIR}/deps/iot-sdk-c ${PROJECT_SOURCE_DIR}/deps/iot-sdk-c -Duse_installed_dependencies=ON -Duse_openssl=OFF -Dbuild_as_dynamic=ON -Dskip_samples=ON -G "${CMAKE_GENERATOR}")
-- 
1.9.1

