From 2d9c5a5006bdbe96b6e73a67d8df7341495d272d Mon Sep 17 00:00:00 2001
From: Scott Ware <scott.r.ware@intel.com>
Date: Wed, 23 Aug 2017 16:20:00 +0100
Subject: [PATCH 5/6] Use shared openssl

Signed-off-by: Scott Ware <scott.r.ware@intel.com>
---
 bindings/nodejs/CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/bindings/nodejs/CMakeLists.txt b/bindings/nodejs/CMakeLists.txt
index 96c4984..0b1582f 100644
--- a/bindings/nodejs/CMakeLists.txt
+++ b/bindings/nodejs/CMakeLists.txt
@@ -30,7 +30,7 @@ if(${LINUX})
         pthread
         dl
         uv
-        openssl
+        ssl
     )
 endif()
 
-- 
1.9.1

